roachtest: specify num replicas

tbg opened the pull request:
We got stricter about when num_replicas needed to be specified, see
b6f8b3dacfe6966e775062efc745fd8f24728fe1.

Touches #41028.

Release justification: roachtest-only fix.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/41384)
<!-- Reviewable:end -->

tbg commented on pkg/workload/tpcc/partition.go:
```diff
@@ -393,7 +393,7 @@ func replicateItem(db *gosql.DB, cfg zoneConfig) error {
 		configure := fmt.Sprintf(`
 			ALTER INDEX item@%s
-			CONFIGURE ZONE USING constraints = '{"+zone=%s":1}', lease_preferences = '[[+zone=%s]]'`,
+			CONFIGURE ZONE USING num_replicas = 3, constraints = '{"+zone=%s":1}', lease_preferences = '[[+zone=%s]]'`,
```
Is `3` the right number here?

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/41384#-:-LqawLFE9w179iiYZEzO:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten and @tbg)

___
*[pkg/workload/tpcc/partition.go, line 396 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/41384#-LqakvFTyoWQdXCOCo_B-r1-396:-LqavywI8vKlvXkPeEQj:bhim0d3) ([raw file](https://github.com/cockroachdb/cockroach/blob/d83de9856cca8504c3144f6becea2e8b22e9ad94/pkg/workload/tpcc/partition.go#L396)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Is `3` the right number here?
</blockquote></details>

Yes, although it would be more general to do `num_replicas = COPY FROM PARENT, `. Up to you.


<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: specify num replicas

We got stricter about when num_replicas needed to be specified, see
b6f8b3dacfe6966e775062efc745fd8f24728fe1.

Touches #41028.

Release justification: roachtest-only fix.

Release note: None

tbg commented:
Using COPY FROM PARENT now.

bors r=nvanbenschoten
